docs/why.md

```diff
@@ -0,0 +1,15 @@
+# Why a monorepo?
+
+Prior to November 2022, this repo was a monolithic repo, where the code for many [controllers](./what.md) lived under one roof and was published as one package (`@metamask/controllers`). This led to a few problems:
+
+- Even if a product team wanted to use a small subset of controllers, they were forced to add the entire package to their project, which bloated the dependency tree unnecessarily.
+- Occasionally, a product team tasked with carving out a new feature needed to make temporary modifications to one or more controllers during development, but this was difficult, as they were forced to fork the repo and keep it up to date.
+- Beyond the controllers repo, we were maintaining code for other shared libraries in separate repos, and keeping these repos architecturally aligned with each other had grown to be painful.
+
+We aimed to improve these situations by splitting up `@metamask/controllers` into many packages, where each package corresponded to a single controller (except for a few cases). By doing this:
+
+- Product teams could now choose which controllers they wanted to use in their project without the fear of relying on more dependencies than were necessary.
+- Instead of forking, product teams can publish [preview builds](./preview-builds.md) for in-progress work and make use of those builds in products to test them out as they are working on new features.
```

Comment:
Just for my education: Why was this facilitated by breaking apart the packages? What made that unviable for the monopackage?

Reply:
I'm not sure about this sentence. Previously, controller forks weren't (to my knowledge) used for experiments that often. They were used by mobile as a medium-term way to introduce controller changes quickly, bypassing the need to make that change functional for the extension and agreeable for other teams.

Experiments were done by installing deps from git references. Which was made impractical due to the Yarn v3 `workspaces: ` protocol.

Reply:
What we could list here as a benefit is that we've made it easier to fork individual controllers without everything else, reducing the maintenance burden of those forks.

Reply:
Yeah, this sentence doesn't make any sense as a motivation, does it 😕  Admittedly, I editorialized the monorepo kickoff document a bit. Given that we aren't able to link to that document anyway I think I will just steal what you wrote @Gudahtt. Maybe the part about forking still makes sense but I'll work it in more as a benefit than a motivation as you said.